build/cmake/CCache.cmake

```diff
@@ -12,11 +12,29 @@
 find_program (CCACHE_EXECUTABLE ccache)
 if (CCACHE_EXECUTABLE)
```

Comment:
I would recommend adding `AND NOT DEFINED MONGO_USE_CCACHE` so that this branch only executes the first time and only if the user hasn't already provided a setting. This also prevents the message() from printing multiple times and possibly not reflecting the value that the user has set.